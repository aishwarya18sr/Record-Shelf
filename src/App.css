.appContainer {
    display: flex;
    flex-direction: column;
    width: 100vw;
    height: 100vh;
}

.appBodyContainer {
    display: flex;
    flex-direction: column;
    margin-top: 60px;
    width: 100vw;
    height: 100%;
    overflow: scroll;
}