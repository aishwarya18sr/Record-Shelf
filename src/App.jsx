import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import HomePage from './pages/HomePage/HomePage';
import SongsPage from './pages/SongsPage/SongsPage';
import GenresPage from './pages/GenresPage/GenresPage';
import { HOME_ROUTE, SONGS_ROUTE, GENRES_ROUTE } from './constants/routes';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route
            path={HOME_ROUTE}
            element={<HomePage />}
          />
          <Route
            path={SONGS_ROUTE}
            element={<SongsPage />}
          />
          <Route
            path={GENRES_ROUTE}
            element={<GenresPage />}
          />
          <Route path="*" element={<div>404 Error. Page not found</div>} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
