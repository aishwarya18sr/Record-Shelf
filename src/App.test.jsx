import { render, screen } from '@testing-library/react';
import React from 'react';
import App from './App';

describe('App', () => {
  it('should render the home page content correctly', () => {
    render(<App />);
    expect(screen.getByText(':((')).toBeTruthy();
    expect(screen.getByText('seems a bit empty in here..')).toBeTruthy();
    expect(screen.getByText('sync')).toBeTruthy();
  });
});
