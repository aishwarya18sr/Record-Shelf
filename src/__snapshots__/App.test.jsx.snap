// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<DocumentFragment>
  <div
    class="appContainer"
  >
    <header
      class="recordsHeader"
    >
      <p
        class="recordsHeaderText"
        data-testid="records-header-text"
      >
        My
        <span
          class="recordsHeaderSpan"
        >
          Record
        </span>
        Shelf
      </p>
    </header>
    <div
      class="appBodyContainer"
    >
      <div
        class="homePageContainer"
      >
        <p
          class="homePageText"
        >
          :((
        </p>
        <p
          class="homePageText"
        >
          seems a bit empty in here..
        </p>
        <button
          class="homePageButton"
          type="button"
        >
          sync
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
